_posts/2017-08-15-Functional_programming.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: post
+title: "Introduction to Functional Programming"
+date: 2017-08-15
+author: Abhay Pratap Singh
+category: Functional Programming
+tags:
+- Functional Programming
+image:
+    url: http://functionaltalks.org/images/site-logo.png
+---
+### What is Functional Programming?
+According to wikipedia, it is defined as:  
+"In computer science, functional programming is a programming paradigm—a style of building the structure and elements of computer programs—that treats computation as the evaluation of mathematical functions and avoids changing-state and mutable data."  
+Didn't get it? No problem! Continue reading and at the end, atleast you will get to know what it is all about.  
+
+Actually Functional Programming is much about abstraction and mathematical concepts. And that is what acts as an hindrance for programmers from learning it. But now many people are turning towards it due to many reasons, primarily the easiness of writing thread-safe(concurrent) code in functional languages. With that in mind, I have tried to cover some key ideas in this post, and have listed some resources at the last. Hoping it will help you! :)  
```

Comment:
The main reason for using functional is to help you write modular and compositional code as modular design is the key to successful programming. (From Hughes paper "Why functional programming matters"). Also try to relate it to how coding jobs require people who write such code.